Java 21 and Gradle

### Before creating

- [X] This is NOT a gameplay feature from Civ VI, BNW, or outside - see Roadmap
- [X] This is NOT a gameplay feature from Vanilla Civ V or from G&K - If so, it should be a comment in https://github.com/yairm210/Unciv/issues/4697

### Problem Description

Currently, on some machines, if you do nothing except download Studio and download the code, you cannot build the code. You will instead get the following:
  ```
 Your build is currently configured to use incompatible Java 21.0.3 and Gradle 8.2. Cannot sync the project.

We recommend upgrading to Gradle version 8.9.

The minimum compatible Gradle version is 8.5.

The maximum compatible Gradle JVM version is 19.

Possible solutions:
 - Upgrade to Gradle 8.9 and re-sync
 - Upgrade to Gradle 8.5 and re-sync
  ```

There's also the stack trace I can paste in, but I double much of it is useful since we probably should update gradle anyways

<details>
  <summary></summary>
  
```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':buildSrc:compileKotlin'.
Caused by: org.gradle.api.internal.tasks.properties.PropertyEvaluationException: Error while evaluating property 'compilerOptions.jvmTarget' of task ':buildSrc:compileKotlin'.
Caused by: org.gradle.api.internal.provider.AbstractProperty$PropertyQueryException: Failed to calculate the value of property 'jvmTarget'.
Caused by: java.lang.IllegalArgumentException: Unknown Kotlin JVM target: 21
	at org.jetbrains.kotlin.gradle.dsl.JvmTarget$Companion.fromTarget(JvmTarget.kt:26)
	at org.jetbrains.kotlin.gradle.tasks.DefaultKotlinJavaToolchain$wireJvmTargetToToolchain$1$1.invoke(DefaultKotlinJavaToolchain.kt:78)
	at org.jetbrains.kotlin.gradle.tasks.DefaultKotlinJavaToolchain$wireJvmTargetToToolchain$1$1.invoke(DefaultKotlinJavaToolchain.kt:70)
        ...
```
  
</details>

--- a/core/build.gradle.kts
+++ b/core/build.gradle.kts
@@ -10,8 +10,6 @@ sourceSets {
 }
 
 kotlin {
-    jvmToolchain(17)
-
     target {
         compilations.all {
             kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()

--- a/desktop/build.gradle.kts
+++ b/desktop/build.gradle.kts
@@ -13,8 +13,6 @@ sourceSets {
 }
 
 kotlin {
-    jvmToolchain(17)
-
     target {
         compilations.all {
             kotlinOptions.jvmTarget = JavaVersion.VERSION_1_8.toString()

